project/Socco.scala

```diff
@@ -0,0 +1,205 @@
+import sbt.Keys._
+import sbt._
+import sbt.plugins.JvmPlugin
+import com.typesafe.sbt.site.SitePlugin
+import java.io.PrintWriter
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.stream.Collectors
+import scala.collection.JavaConverters._
+import scala.util.Try
+import scala.io.Source
+
+object SbtSoccoPlugin extends AutoPlugin {
```

Comment:
feel like we could go even simpler and just leave this as a plain object like:

```scala
object Socco {
  val settings = Seq(
     commands += enableSoccoCommand,
     soccoOnCompile.in(p) := onCompile,
      soccoPackage := Nil,
      soccoOut := target.value / "socco",
      soccoIndex := index.value,
      ...
     )
}
```

and just add `.settings(Socco.settings)` to the `site` module in build.sbt. wdyt?